src/ReachSets/reach.jl

```diff
@@ -211,6 +214,130 @@ function reach(S::AbstractSystem,
+- `kwargs`             -- (optional) additional arguments that are passed to the
+                          backend
+
+### Notes
+
+A dictionary with available algorithms is available via
+`Reachability.available_algorithms`.
+"""
+function d_reach(HS::HybridSystem, # Does AbstractSystem a superset of HybridSystem?
+               N::Int;
+               X0::BallInf; # TODO is it correct to way to specify initial set for HS?
+               algorithm::String="explicit",
+               ε_init::Float64=Inf,
+               set_type_init::Type{<:LazySet}=Hyperrectangle,
+               ε_iter::Float64=Inf,
+               set_type_iter::Type{<:LazySet}=Hyperrectangle,
+               assume_sparse=true,
+               assume_homogeneous=false,
+               numeric_type::Type=Float64,
+               lazy_X0=false,
+               kwargs...)::Vector{<:LazySet}
+               # unpack arguments
+               kwargs_dict = Dict(kwargs)
+
+
+
+               # list containing the arguments passed to any reachability function
+               args = []
+               waiting_list = []
+               # coefficients matrix
+               #TODO get start state. For now we assume that it is the first location
+               cur_loc_id = 1
+               cur_loc = HS.modes[init_loc_id]
+
+               #TODO push_initial_set
+               push!((cur_loc[1], X0), waiting_list)
+               i = 0
+               while (!isempty(waiting_list) and i < 15) #TODO add variable for max iteration number
+
+                   S = ContinuousSystem(cur_loc.A, X0, cur_loc.U)
+
+                   Rsets = reach(
+                        S,
+                        kwargs_dict[:N];
+                        algorithm=kwargs_dict[:algorithm],
+                        ε_init=kwargs_dict[:ε_init],
+                        set_type_init=kwargs_dict[:set_type_init],
+                        ε_iter=kwargs_dict[:ε_iter],
+                        set_type_iter=kwargs_dict[:set_type_iter],
+                        assume_sparse=kwargs_dict[:assume_sparse],
+                        assume_homogeneous=kwargs_dict[:assume_homogeneous],
+                        lazy_X0=kwargs_dict[:lazy_X0],
+                        blocks=kwargs_dict[:blocks],
+                        partition=kwargs_dict[:partition],
+                        block_types_init=kwargs_dict[:block_types_init],
+                        block_types_iter=kwargs_dict[:block_types_iter],
+                        template_directions_init=kwargs_dict[:template_directions_init],
+                        template_directions_iter=kwargs_dict[:template_directions_iter],
+                        lazy_inputs_interval=kwargs_dict[:lazy_inputs_interval],
+                        output_function=kwargs_dict[:output_function]
+                        )
+                       for (j, tj) in enumerate(transitions(HS)) #TODO optimize it or add method for outgoing transitions to SX
```

Comment:
indentation!
@mforets: Is there no way to retrieve the outgoing transitions of location `cur_loc`? We definitely want to have this!

Reply:
this is probably https://blegat.github.io/HybridSystems.jl/latest/lib/methods.html#HybridSystems.out_transitions

Reply:
Thanks! Fixed according to this API